### ISSUE TITLE
Invitation handling is not working after user accepted invitation.

### ISSUE THREAD

**Issue by dpasek-senacor** [mongodb/terraform-provider-mongodbatlas#945]
### Terraform CLI and Terraform MongoDB Atlas Provider Version
Using Terraform Cloud VCS Workflow for Applying state changes.
```hcl
Terraform v1.3.5
+ provider registry.terraform.io/mongodb/mongodbatlas v1.6.0
```

### Terraform Configuration File 
```hcl
terraform {
  cloud {
    organization = "tf-mongo-org"
    workspaces {
      name = "mongo-organization"
    }
  }
}

terraform {
  required_providers {
    mongodbatlas = {
      source = "mongodb/mongodbatlas"
    }
  }
}

provider "mongodbatlas" {
  public_key = var.public_key
  private_key = var.private_key
}

resource "mongodbatlas_org_invitation" "invitation" {  
  username = "test@myorg.com"
  org_id = var.mongo_org_id
  roles = ["ORG_MEMBER"]
}
```

### Steps to Reproduce
1. Perform Plan&Apply on Terraform Cloud
2. Users accepts invitations email
3. Perform Plan&Apply on Terraform Cloud
4. Remove `mongodbatlas_org_invitation.invitation` from configuration
5. Perform Plan&Apply on Terraform Cloud

### Expected Behavior
Step 3: It would be expected that the state is updated in the way that the invitation is removed from Terraform state.
Step 5: It would be expected that the plan does not detected any changes as state has been already been updated or the removal of invitation from configuration is working if state still contains the resource.

### Actual Behavior
Step 3: The Terraform state is not updated. No state changes are detected during plan or apply and the invitation is still part of the Terraform state afterwards. Even a "Refresh State" (aka `terraform apply -refresh-only`) does not detect that the invitation is no longer active.
Step 5: The Plan phase shows a deletion of the invitation . Applying the invitation fails as the resource cannot be deleted as it is not found (HTTP 404)

### Debug Output 
Debug output of trying to delete the resource in Step 5 (see above)
```plain
2022-11-30T14:12:23.870Z [INFO]  provider.terraform-provider-mongodbatlas_v1.6.0: 2022/11/30 14:12:23 [DEBUG] MongoDB Atlas API Request Details:
---[ REQUEST ]---------------------------------------
DELETE /api/atlas/v1.0/orgs/63839ba5b41c43346b8708ff/invites/6384f323dd49394ed60495d1 HTTP/1.1
Host: cloud.mongodb.com
User-Agent: terraform-provider-mongodbatlas/1.6.0 go-mongodbatlas/0.18.0 (linux;amd64)
Accept: application/json
Accept-Encoding: gzip


-----------------------------------------------------: timestamp=2022-11-30T14:12:23.870Z
2022-11-30T14:12:23.918Z [INFO]  provider.terraform-provider-mongodbatlas_v1.6.0: 2022/11/30 14:12:23 [DEBUG] MongoDB Atlas API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 404 Not Found
Content-Length: 188
Content-Type: application/json
Date: Wed, 30 Nov 2022 14:12:23 GMT
Referrer-Policy: strict-origin-when-cross-origin
Server: envoy
Strict-Transport-Security: max-age=31536000; includeSubdomains;
X-Content-Type-Options: nosniff
X-Envoy-Upstream-Service-Time: 23
X-Frame-Options: DENY
X-Mongodb-Service-Version: gitHash=6f960f6384e938f185ba29e3a46877f5157a4d6b; versionString=v20221116
X-Permitted-Cross-Domain-Policies: none

{
"detail": "An invalid invitation ID 6384F323DD49394ED60495D1 was specified.",
"error": 404,
"errorCode": "INVALID_INVITATION_ID",
"parameters": [
],
"reason": "Not Found"
}
-----------------------------------------------------: timestamp=2022-11-30T14:12:23.918Z
2022-11-30T14:12:23.919Z [ERROR] provider.terraform-provider-mongodbatlas_v1.6.0: Response contains error diagnostic: @module=sdk.proto diagnostic_detail= tf_proto_version=5.3 tf_resource_type=mongodbatlas_org_invitation @caller=github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/internal/diag/diagnostics.go:55 tf_provider_addr=provider tf_req_id=4de73b68-3926-9005-0934-7df666e3533a tf_rpc=ApplyResourceChange diagnostic_severity=ERROR diagnostic_summary="error deleting Organization invitation for user test@myorg.com: DELETE https://cloud.mongodb.com/api/atlas/v1.0/orgs/63839ba5b41c43346b8708ff/invites/6384f323dd49394ed60495d1: 404 (request "INVALID_INVITATION_ID") An invalid invitation ID 6384F323DD49394ED60495D1 was specified." timestamp=2022-11-30T14:12:23.919Z
```

### References
https://github.com/mongodb/terraform-provider-mongodbatlas/issues/636

**Comment by dpasek-senacor** [comment]
Here is additional log output from performing a "refresh state" after the user has accepted the invitation:
```plain
2022-11-30T16:16:19.308Z [DEBUG] Building and walking plan graph for RefreshOnlyMode
2022-11-30T16:16:19.308Z [DEBUG] ProviderTransformer: "mongodbatlas_org_invitation.invite (expand)" (*terraform.nodeExpandPlannableResource) needs provider["registry.terraform.io/mongodb/mongodbatlas"]
2022-11-30T16:16:19.308Z [DEBUG] ReferenceTransformer: "mongodbatlas_org_invitation.invite (expand)" references: [var.mongodb_org_id]
2022-11-30T16:16:19.308Z [DEBUG] ReferenceTransformer: "var.mongodb_org_id" references: []
2022-11-30T16:16:19.308Z [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/mongodb/mongodbatlas\"]" references: []
2022-11-30T16:16:19.309Z [DEBUG] Starting graph walk: walkPlan
2022-11-30T16:16:19.309Z [DEBUG] created provider logger: level=debug
2022-11-30T16:16:19.310Z [INFO]  provider: configuring client automatic mTLS
2022-11-30T16:16:19.316Z [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/mongodb/mongodbatlas/1.6.0/linux_amd64/terraform-provider-mongodbatlas_v1.6.0 args=[.terraform/providers/registry.terraform.io/mongodb/mongodbatlas/1.6.0/linux_amd64/terraform-provider-mongodbatlas_v1.6.0]
2022-11-30T16:16:19.321Z [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/mongodb/mongodbatlas/1.6.0/linux_amd64/terraform-provider-mongodbatlas_v1.6.0 pid=88
2022-11-30T16:16:19.321Z [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/mongodb/mongodbatlas/1.6.0/linux_amd64/terraform-provider-mongodbatlas_v1.6.0
2022-11-30T16:16:19.335Z [INFO]  provider.terraform-provider-mongodbatlas_v1.6.0: configuring server automatic mTLS: timestamp=2022-11-30T16:16:19.335Z
2022-11-30T16:16:19.353Z [DEBUG] provider.terraform-provider-mongodbatlas_v1.6.0: plugin address: address=/tmp/plugin40202895 network=unix timestamp=2022-11-30T16:16:19.353Z
2022-11-30T16:16:19.353Z [DEBUG] provider: using plugin: version=5
2022-11-30T16:16:19.401Z [WARN]  ValidateProviderConfig from "provider[\"registry.terraform.io/mongodb/mongodbatlas\"]" changed the config value, but that value is unused
2022-11-30T16:16:19.402Z [INFO]  ReferenceTransformer: reference not found: "var.mongodb_org_id"
2022-11-30T16:16:19.403Z [DEBUG] ReferenceTransformer: "mongodbatlas_org_invitation.invite" references: []
2022-11-30T16:16:19.407Z [INFO]  provider.terraform-provider-mongodbatlas_v1.6.0: 2022/11/30 16:16:19 [DEBUG] MongoDB Atlas API Request Details:
---[ REQUEST ]---------------------------------------
GET /api/atlas/v1.0/orgs/<ORG_ID>/invites/63877f725b60881b1edba60e HTTP/1.1
Host: cloud.mongodb.com
User-Agent: terraform-provider-mongodbatlas/1.6.0 go-mongodbatlas/0.18.0 (linux;amd64)
Accept: application/json
Accept-Encoding: gzip


-----------------------------------------------------: timestamp=2022-11-30T16:16:19.407Z
2022-11-30T16:16:19.450Z [INFO]  provider.terraform-provider-mongodbatlas_v1.6.0: 2022/11/30 16:16:19 [DEBUG] MongoDB Atlas API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 404 Not Found
Content-Length: 188
Content-Type: application/json
Date: Wed, 30 Nov 2022 16:16:19 GMT
Referrer-Policy: strict-origin-when-cross-origin
Server: envoy
Strict-Transport-Security: max-age=31536000; includeSubdomains;
X-Content-Type-Options: nosniff
X-Envoy-Upstream-Service-Time: 21
X-Frame-Options: DENY
X-Mongodb-Service-Version: gitHash=6f960f6384e938f185ba29e3a46877f5157a4d6b; versionString=v20221116
X-Permitted-Cross-Domain-Policies: none

{
"detail": "An invalid invitation ID 63877F725B60881B1EDBA60E was specified.",
"error": 404,
"errorCode": "INVALID_INVITATION_ID",
"parameters": [
],
"reason": "Not Found"
}
-----------------------------------------------------: timestamp=2022-11-30T16:16:19.449Z
2022-11-30T16:16:19.450Z [INFO]  provider.terraform-provider-mongodbatlas_v1.6.0: 2022/11/30 16:16:19 [DEBUG] MongoDB Atlas API Request Details:
---[ REQUEST ]---------------------------------------
GET /api/atlas/v1.0/users/byName/test@myorg.com HTTP/1.1
Host: cloud.mongodb.com
User-Agent: terraform-provider-mongodbatlas/1.6.0 go-mongodbatlas/0.18.0 (linux;amd64)
Accept: application/json
Accept-Encoding: gzip


-----------------------------------------------------: timestamp=2022-11-30T16:16:19.450Z
2022-11-30T16:16:19.509Z [INFO]  provider.terraform-provider-mongodbatlas_v1.6.0: 2022/11/30 16:16:19 [DEBUG] MongoDB Atlas API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 200 OK
Content-Type: application/json
Date: Wed, 30 Nov 2022 16:16:19 GMT
Referrer-Policy: strict-origin-when-cross-origin
Server: envoy
Strict-Transport-Security: max-age=31536000; includeSubdomains;
Vary: Accept-Encoding
X-Content-Type-Options: nosniff
X-Envoy-Upstream-Service-Time: 51
X-Frame-Options: DENY
X-Mongodb-Service-Version: gitHash=6f960f6384e938f185ba29e3a46877f5157a4d6b; versionString=v20221116
X-Permitted-Cross-Domain-Policies: none

{
"country": "DE",
"createdAt": "2022-11-01T12:50:10Z",
"emailAddress": "test@myorg.com",
"firstName": "John",
"id": "<USER_ID>",
"lastAuth": "2022-11-30T16:08:17Z",
"lastName": "Doe",
"links": [
{
"href": "https://cloud.mongodb.com/api/atlas/v1.0/users/<USER_ID>",
"rel": "self"
},
{
"href": "https://cloud.mongodb.com/api/atlas/v1.0/users/<USER_ID>/whitelist",
"rel": "https://cloud.mongodb.com/whitelist"
},
{
"href": "https://cloud.mongodb.com/api/atlas/v1.0/users/<USER_ID>/accessList",
"rel": "https://cloud.mongodb.com/accessList"
}
],
"mobileNumber": "",
"roles": [
{
"orgId": "<ORG_ID>",
"roleName": "ORG_MEMBER"
}
],
"teamIds": [],
"username": "test@myorg.com"
}
-----------------------------------------------------: timestamp=2022-11-30T16:16:19.509Z
2022-11-30T16:16:19.511Z [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2022-11-30T16:16:19.512Z [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/mongodb/mongodbatlas/1.6.0/linux_amd64/terraform-provider-mongodbatlas_v1.6.0 pid=88
2022-11-30T16:16:19.512Z [DEBUG] provider: plugin exited
2022-11-30T16:16:19.512Z [DEBUG] no planned changes, skipping apply graph check
2022-11-30T16:16:19.512Z [INFO]  backend/local: plan operation completed
2022-11-30T16:16:19.512Z [INFO]  backend/local: writing plan output to: /home/tfc-agent/.tfc-agent/component/terraform/runs/run-tCmmRRYEQTD7FFLz/config/terraform.tfplan
```
As I can see the invitation is loaded causing a 404 and then the provider validates if the user is now part of the organisation (which is the case) but does not remove the invitation from the state.

**Comment by martinstibbe** [comment]
@dpasek-senacor Confirmed this is a bug will fix in engineering ticket [INTMDB-511](https://jira.mongodb.org/browse/INTMDB-511)

**Comment by vlsokol** [comment]
Do we have an ETA on when this can be picked up?

**Comment by Zuhairahmed** [comment]
Q1 2023

**Comment by Zuhairahmed** [comment]
@dpasek-senacor @vlsokol as an update, this issue has been picked up in current sprint cycle and you can expect fix to be included as part of v1.8.0 release next week. will let you know when latest release has been published to the Terraform Registry.

**Comment by Zuhairahmed** [comment]
@dpasek-senacor @vlsokol we just released v1.8.0, feel free to give it try! issue should have been resolved.

**Comment by Zuhairahmed** [comment]
closing this issue as you appear to be unblocked, but feel free to re-open if you need anything else
